### PR TITLE
Add chart CR support to collector

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/5561

This is prep for adding an alert for incorrect tiller config. Currently the collector only supports chartconfig CRs this extends it to also consider chart CRs.

### CODE REVIEW

File: service/collector/chart_resource.go [hunk 1]
```diff
@@ -55,14 +55,30 @@ func (c *Collector) collectChartConfigStatus(ctx context.Context, ch chan<- prom
 			chartConfig.channelName,
 			chartConfig.releaseName,
 			chartConfig.releaseStatus,
-			defaultNamespace,
+			c.watchNamespace,
```
Comment (rossf7) [inline service/collector/chart_resource.go]: Removing the hardcoding of the namespace.

File: service/collector/tiller_configured.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		prometheus.GaugeValue,
+		value,
+		c.tillerNamespace,
+	)
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "finished collecting Tiller configuration")
+}
+
+func (c *Collector) checkTillerDeployment() error {
+	deploy, err := c.k8sClient.Extensions().Deployments(c.tillerNamespace).Get("tiller-deploy", metav1.GetOptions{})
```
Comment (kopiczko) [inline service/collector/tiller_configured.go]: Does it make sense to put `"tiller-deploy"` into `key` package?

File: service/collector/tiller_configured.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	}
+
+	for _, envVar := range containers[0].Env {
+		if envVar.Name == "TILLER_HISTORY_MAX" && envVar.Value == "10" {
+			// Configuration is correct.
+			return nil
+		}
+	}
+
+	return microerror.Maskf(invalidExecutionError, "tiller configuration is incorrect")
```
Comment (kopiczko) [inline service/collector/tiller_configured.go]: I'd add that we expected to have `TILLER_HISTORY_MAX=10`.

File: service/collector/tiller_configured.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+)
+
+func (c *Collector) collectTillerConfigured(ctx context.Context, ch chan<- prometheus.Metric) {
+	var value float64
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting Tiller configuration")
+
+	charts, err := c.getCharts()
+	if err != nil {
+		c.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("could not get Charts"), "stack", fmt.Sprintf("%#v", err))
```
[the remark this answers is not in the document]
Comment (rossf7) [reply service/collector/tiller_configured.go]: @kopiczko Sorry this file should have been added in the next PR :(

I'll make sure the feedback is addressed there.

File: service/collector/tiller_configured.go (hunk 2 above)
Comment (rossf7) [reply service/collector/tiller_configured.go]: Yes good shout.

File: service/collector/tiller_configured.go (hunk 3 above)
Comment (rossf7) [reply service/collector/tiller_configured.go]: Same for this.

File: service/collector/tiller_configured.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+		return microerror.Mask(err)
+	}
+
+	containers := deploy.Spec.Template.Spec.Containers
+	if len(containers) != 1 {
+		return microerror.Maskf(invalidExecutionError, "tiller container not found expected 1 got %d", len(containers))
+	}
+
+	for _, envVar := range containers[0].Env {
+		if envVar.Name == "TILLER_HISTORY_MAX" && envVar.Value == "10" {
```
Comment (tomahawk28) [inline service/collector/tiller_configured.go]: Can we move it this `TILLER_HISTORY_MAX` and `10` as const value?

File: service/collector/tiller_configured.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+		prometheus.BuildFQName(Namespace, "", "tiller_configured"),
+		"Tiller is configured correctly.",
+		[]string{
+			namespaceLabel,
+		},
+		nil,
+	)
+)
+
+func (c *Collector) collectTillerConfigured(ctx context.Context, ch chan<- prometheus.Metric) {
```
Comment (tomahawk28) [inline service/collector/tiller_configured.go]: Even if it's a private method, I don't think it's easy to find out what is the purpose of this method.

**Comment by rossf7** [comment]
Tested on ginger.

**Pull request merged**